File: chapter2/typescript/ShapeFactory.ts
import { Circle } from "./Circle";
import { Oval } from "./Oval";
import { Shape } from "./Shape";
import { Square } from "./Square";

/**
 * Factory Pattern - ShapeFactory
 *
 * @author: bridget
 */
 export class ShapeFactory {
    static createShape(shapeType: String): Shape {
        if ("CIRCLE"==shapeType) {
            return new Circle();

        } else if ("OVAL"== shapeType) {
            return new Oval();

        } else if ("SQUARE" == shapeType) {
            return new Square();
        }
        throw new Error("Shape is not supported: " + shapeType);
    }
}

File: chapter2/java/designpattern/src/main/java/com/designpattern/ShapeFactory.java
package com.designpattern;

/**
 * Factory Pattern - ShapeFactory
 *
 * @author: bridget
 */
public class ShapeFactory {
    public Shape createShape(String shapeType) {
        if ("CIRCLE".equals(shapeType)) {
            return new Circle();

        } else if ("OVAL".equals(shapeType)) {
            return new Oval();

        } else if ("SQUARE".equals(shapeType)) {
            return new Square();
        }
        throw new UnsupportedOperationException("Shape is not supported: " + shapeType);
    }
}
